src/Sfa.Tl.Matching.Data/Repositories/OpportunityRepository.cs

```diff
@@ -192,8 +203,14 @@ where IsValidBasketState(oi, OpportunityType.Referral)
                                           DistanceFromEmployer = re.DistanceFromEmployer,
                                           PlacementsKnown = oi.PlacementsKnown,
                                           Placements = oi.Placements,
-                                          ProviderName = pv.Name == pv.Postcode ? p.DisplayName : $"{pv.Name} (part of {p.DisplayName})",
+                                          ProviderName = ProviderDisplayExtensions.GetDisplayText(pv.Name, pv.Postcode, p.DisplayName),
```

Comment:
Same as ProviderVenue, put this in to mapper or Dto

Reply:
That will also simplify the generated SQL

Reply:
Updated